workspace.spec

```diff
@@ -572,6 +572,378 @@ module Workspace {
 		string description;
 	} ProvenanceAction;
 
+	/* Credit Metadata-related objects and fields
+
+		Workspace objects have a list of zero or more sets of credit metadata (CM), or citation information for the object. This information allows data imported into KBase to retain the appropriate details for users wishing to cite the data source, and for those who contributed to its creation to be credited for their work.
+
+		CM is stored in a list as part of the object provenance, but unlike other provenance information, which is immutable, CM can be updated.
+
+		Credit metadata is added after object creation, and requires the user to be a workspace admin. Username, timestamp, and credit metadata schema version are stored along with the CM, which allows tracking of CM changes.
+
+		To update existing CM, a new set of credit metadata is added to the list.
+
+		When requesting credit metadata, the last (i.e. most recent) entry in the list is returned.
+
+		In the following entries, "resource" is used to refer to the workspace object that the CM pertains to.
+	*/
+
+	/* EventDate
+
+		Represents an event in the lifecycle of a resource and the date it occurred on.
+
+		See https://support.datacite.org/docs/datacite-metadata-schema-v44-recommended-and-optional-properties#8-date for more information on the events (below).
+
+		Both event and date are required fields.
+
+		string date - the date associated with the event. The date may be in the format
```

Comment:
I've started moving away from recapitulating the type in the comment - seems redundant. Up to you whether you want to do that here. Otherwise, at some point in the future, I'll probably do a clean sweep of the spec

Reply:
I'll remove them all in the next commit.

Reply:
If it's just the types for your additions that great, but if you're talking about the entire spec could we please do that in a separate PR?

Reply:
Done for the entries in this PR.

Reply:
:+1: